File: C++/easy/242_valid_anagram.cc
#include <iostream>

using namespace std;

class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        int m[26] = {0};

        for (char c : s)
            m[c - 'a']++;

        for (char c : t)
            m[c - 'a']--;

        for (int i : m)
            if (i != 0)
                return false;

        return true;
    }
};

File: Python/easy/242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        values = {}
        for value in s:
            if value in values:
                values[value] += 1
            else:
                values[value] = 1

        for value in t:
            res = values.get(value)
            if not res:
                return False

            values[value] -= 1

        for key in values:
            if values[key] != 0:
                return False

        return True
